### DISCUSSION TITLE
The comparison result of Unicode strings looks strange...

### DISCUSSION THREAD

**Discussion by Flithor** [dotnet/runtime#51726] [Q&A] [answered]
```
//一: Chinese character, meaning the number 1
//二: Chinese character, meaning the number 2
String.Compare("一", "二")
//-> 1 //"一" is bigger than "二"
string.Compare("一", "二", false, System.Globalization.CultureInfo.GetCultureInfo("zh-CN"))
//-> 1 //"一" is bigger than "二" under culture of "zh-CN"
```
But...
```
(int)'一'
//-> 19968
(int)'二'
//-> 20108
'一' > '二'
//-> false // '一' is smaller than '二'
```
Why is that?

This is a bug or WAD?

**Comment by huoyaoyuan** [comment, marked as answer]
The default overload of string.Compare uses current culture. Under Chinese culture, the comparison is based on Pinyin, thus yi > er.

**Reply by Flithor** [reply]
Uh... well

**Comment by Clockwork-Muse** [comment]
I can't speak to what the comparison result _should_ be, other than to note that sorting solely on unicode codepoint is probably not correct, since there are multiple languages that use Latin characters plus some additions (say, Turkish).  Unicode doesn't redefine the Latin characters, and if a specific order is to be retained you need some custom sort table.
